packages/earn-controller/src/EarnController.test.ts

```diff
@@ -2566,4 +2566,271 @@ describe('EarnController', () => {
+        Object {
+          "lastUpdated": 0,
+          "lending": Object {
+            "isEligible": true,
+            "markets": Array [
+              Object {
+                "address": "0xe50fa9b3c56ffb159cb0fca61f5c9d750e8128c8",
+                "chainId": 42161,
+                "id": "0xe50fa9b3c56ffb159cb0fca61f5c9d750e8128c8",
+                "name": "0xe50fa9b3c56ffb159cb0fca61f5c9d750e8128c8",
+                "netSupplyRate": 1.52269127978874,
+                "outputToken": Object {
+                  "address": "0xe50fa9b3c56ffb159cb0fca61f5c9d750e8128c8",
+                  "chainId": 42161,
+                },
+                "protocol": "aave",
+                "rewards": Array [],
+                "totalSupplyRate": 1.52269127978874,
+                "tvlUnderlying": "132942564710249273623333",
+                "underlying": Object {
+                  "address": "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
+                  "chainId": 42161,
+                },
+              },
+            ],
+            "positions": Array [
+              Object {
+                "assets": "112",
+                "chainId": 42161,
+                "id": "0xe6a7d2b7de29167ae4c3864ac0873e6dcd9cb47b-0x078f358208685046a11c85e8ad32895ded33a249-COLLATERAL-0",
+                "market": Object {
+                  "address": "0x078f358208685046a11c85e8ad32895ded33a249",
+                  "chainId": 42161,
+                  "id": "0x078f358208685046a11c85e8ad32895ded33a249",
+                  "name": "0x078f358208685046a11c85e8ad32895ded33a249",
+                  "netSupplyRate": 0.0062858302613958,
+                  "outputToken": Object {
+                    "address": "0x078f358208685046a11c85e8ad32895ded33a249",
+                    "chainId": 42161,
+                  },
+                  "protocol": "aave",
+                  "rewards": Array [],
+                  "totalSupplyRate": 0.0062858302613958,
+                  "tvlUnderlying": "315871357755",
+                  "underlying": Object {
+                    "address": "0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f",
+                    "chainId": 42161,
+                  },
+                },
+                "marketAddress": "0x078f358208685046a11c85e8ad32895ded33a249",
+                "marketId": "0x078f358208685046a11c85e8ad32895ded33a249",
+                "protocol": "aave",
+              },
+            ],
+          },
+        }
+      `);
+    });
+
+    it('persists expected state', async () => {
+      const { controller } = await setupController();
+
+      const derivedState = deriveStateFromMetadata(
+        controller.state,
+        controller.metadata,
+        'persist',
+      );
+
+      // Compare `pooled_staking` separately to minimize size of snapshot
+      const {
+        pooled_staking: derivedPooledStaking,
+        ...derivedStateWithoutPooledStaking
+      } = derivedState;
+      expect(derivedPooledStaking).toStrictEqual({
```

Comment:
This is a bit.... janky. It's not as easy to read as I want it to be.

 I've hard-coded the expected values here that get setup in `setupController` from the mocks, because it was _way_ too large for an inline snapshot. And since those mocks are setup in the root `describe` block, there isn't an easy way to customize them here.

Ideally the constructor wouldn't do this initialization at all, leaving it to an `init` function. And ideally we'd be able to customize it in `setupController` rather than it being set for all tests. But either of those changes aren't feasible right now.

Please feel free to suggest alternatives, if you can think of any that would be more readable.